app/services/katello/pulp/server.rb

```diff
@@ -4,23 +4,33 @@ class Server
       def self.config(url, user_remote_id)
         uri = URI.parse(url)
 
-        Runcible::Instance.new(
-          :url => "#{uri.scheme}://#{uri.host.downcase}",
-          :api_path => uri.path,
-          :user => user_remote_id,
-          :timeout => SETTINGS[:katello][:rest_client_timeout],
-          :open_timeout => SETTINGS[:katello][:rest_client_timeout],
-          :cert_auth => {
-            :ssl_client_cert => ::Cert::Certs.ssl_client_cert,
-            :ssl_client_key => ::Cert::Certs.ssl_client_key
-          },
-          :logging => {
+        # rubocop:disable EmptyLiteral
+        runcible_params = Hash.new.tap do |runc|
+          runc[:url] = "#{uri.scheme}://#{uri.host.downcase}"
+          runc[:api_path] = uri.path
+          runc[:user] = user_remote_id
+          runc[:timeout] = SETTINGS[:katello][:rest_client_timeout]
+          runc[:open_timeout] = SETTINGS[:katello][:rest_client_timeout]
+          if SETTINGS[:katello][:pulp][:use_oauth]
+            runc[:oauth] = {
+              :oauth_secret => SETTINGS[:katello][:pulp][:oauth_secret],
+              :oauth_key => SETTINGS[:katello][:pulp][:oauth_key]
+            }
+          else
+            runc[:cert_auth] = {
+              :ssl_client_cert => ::Cert::Certs.ssl_client_cert,
+              :ssl_client_key => ::Cert::Certs.ssl_client_key
+            }
+          end
+          runc[:logging] = {
             :logger => ::Foreman::Logging.logger('katello/pulp_rest'),
             :exception => true,
             :info => true,
             :debug => true
           }
-        )
+        end
+
+        Runcible::Instance.new(runcible_params)
```

Comment:
While i'm sure this works, its a bit hard to read.  I'd probably prefer something like:
```
runcible_params = {
    :url => "#{uri.scheme}://#{uri.host.downcase}"
    :api_path =>  uri.path
    Other_non_shared_attributes
}

if SETTINGS[:katello][:pulp][:use_oauth]
  runcible_params[:oauth] = {
              :oauth_secret => SETTINGS[:katello][:pulp][:oauth_secret],
              :oauth_key => SETTINGS[:katello][:pulp][:oauth_key]
            }
else
   runcible_params[:cert_auth] = {
        :ssl_client_cert => ::Cert::Certs.ssl_client_cert,
         :ssl_client_key => ::Cert::Certs.ssl_client_key
  }

Runcible::Instance.new(runcible_params)
end
```